### PR TITLE
Add RPM builder for VMware Photon OS

### PR DESCRIPTION
**Pull request by frapposelli**
This is a followup to PR #20108

Photon OS has hit GA a few days back and we'd like to submit this again as #20108 was closed waiting on us being generally avaialable.

There is one enhancement to the build script that no longer relies on an external script to install dependencies.

/cc: @tianon @thaJeztah

### CODE REVIEW

**Comment by runcom** [comment]
LGTM

**Comment by frapposelli** [comment]
@thaJeztah @tianon @runcom any other feedback on this?

**Comment by thaJeztah** [comment]
@frapposelli sorry for the delay; I actually started a discussion on our maintainers list. Let my try to give a  TL;DR - We need to sort out if (and when) we'll include Photon OS in our "official builds", and distribute packages.

We currently don't have clear rules for that, and I think we should get that sorted out as well, for example:

- having CI in place for all packages/platforms we ship
- having it included in our documentation (installation guide)
- be sure that maintainers are able to test (and "support") the distros we ship (and/or have dedicated people upstream to help us with that)

Our current scripts will automatically build all distros in the "contrib/builder" directory, so merging this would automatically result in building packages during release.

We should have this all sorted / decided on before the 1.13 release

**Comment by frapposelli** [comment]
@thaJeztah thanks for the detailed explanation.

From our side we are maintaining an official docker image for Photon OS that is used to build RPMs here, so there shouldn't be any surprises on the building/packaging side.

I understand the challenges, let me know if I can be of any help, you can find me at fabio@vmware.com as well.

**Comment by thaJeztah** [comment]
Thanks @frapposelli! We're currently prioritizing on issues for 1.12, but let me add an "needs attention" label, and put it on the 1.13 milestone. I will keep you informed on progress

**Comment by crosbymichael** [comment]
LGTM

**Comment by crosbymichael** [comment]
@frapposelli can you rebase this and re-generate the Dockerfile?  After that we can merge it.

Thanks!

**Comment by tiborvass** [comment]
@frapposelli hey, I see you're not modifying `hack/install.sh` so I'm just wondering how you intended to distribute docker on photonOS? If I'm a photonOS user, what should i do to install docker? I guess we might need a new section for PhotonOS in the install docs.

**Comment by frapposelli** [comment]
@tiborvass good catch 😄 , I'm amending the PR right now.

File: hack/install.sh [hunk 1]
```diff
@@ -141,7 +141,7 @@ do_install() {
 	esac
 
 	if command_exists docker; then
-		version="$(docker -v | awk -F '[ ,]+' '{ print $3 }')"
+		version="$(docker -v | cut -d ' ' -f3 | cut -d ',' -f1)"
```
Comment (frapposelli) [inline hack/install.sh]: Replaced `awk` with `cut` for better compatibility (`awk` is not installed by default on Photon OS)

cc: @tiborvass

**Comment by frapposelli** [comment]
@crosbymichael @tiborvass amended, let me know what are the next steps wrt documentation.

**Comment by frapposelli** [comment]
rebased

**Comment by frapposelli** [comment]
rebased once again and regenerated `Dockerfile` for `photon:1.0`.

ping @crosbymichael @tiborvass

**Comment by frapposelli** [comment]
ping @crosbymichael @tiborvass @thaJeztah @runcom

**Comment by icecrime** [comment]
LGTM! Ping @tianon :innocent:

**Comment by tianon** [comment]
LGTM :+1:

File: contrib/builder/rpm/amd64/photon-1.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#
+# THIS FILE IS AUTOGENERATED; SEE "contrib/builder/rpm/amd64/generate.sh"!
+#
+
+FROM photon:1.0
+
+RUN tdnf install -y wget curl ca-certificates gzip make rpm-build sed gcc linux-api-headers glibc-devel binutils libseccomp libltdl-devel elfutils
+RUN tdnf install -y btrfs-progs-devel device-mapper-devel glibc-static libseccomp-devel libselinux-devel libtool-ltdl-devel pkg-config selinux-policy selinux-policy-devel sqlite-devel systemd-devel tar git
+
+ENV GO_VERSION 1.6.3
```
Comment (tiborvass) [inline contrib/builder/rpm/amd64/photon-1.0/Dockerfile]: We probably want GO_VERSION to be `1.7` now :)

Comment (frapposelli) [inline contrib/builder/rpm/amd64/photon-1.0/Dockerfile]: done 😄

**Comment by frapposelli** [comment]
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉

**Comment by tiborvass** [comment]
Thanks for your patience @frapposelli :)

**Comment by tiborvass** [comment]
To be clear, this will be uploaded for the next release, so 1.13.0.

**Pull request merged**